### PR TITLE
workaround for executing on Mojave

### LINKED ISSUE #160: prepare script failing for Big Sur
Just grabbed latest prepare-jamf-policy.sh and pointing it at Big Sur, it goes to make the DMG, and then after a few minutes I see:

`Ok, building DMG file of Install macOS Big Sur.app. Wait few minutes.
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Frameworks/OSInstallerSetup.framework/Versions/A/Frameworks/IAESD.framework/Versions/A/Frameworks/IABridgeOSInstall.framework/Versions/A/_CodeSignature/CodeResources: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Frameworks/OSInstallerSetup.framework/Versions/A/Frameworks/IAESD.framework/Versions/A/Frameworks/IABridgeOSInstall.framework/Versions/A/Resources/framework.sb: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Frameworks/OSInstallerSetup.framework/Versions/A/Frameworks/IAESD.framework/Versions/A/Frameworks/IABridgeOSInstall.framework/Versions/A/Resources/Info.plist: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Frameworks/OSInstallerSetup.framework/Versions/A/Frameworks/IAESD.framework/Versions/A/Frameworks/IABridgeOSInstall.framework/Versions/A/Resources/version.plist: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Frameworks/OSInstallerSetup.framework/Versions/A/Frameworks/IAESD.framework/Versions/A/Frameworks/IABridgeOSInstall.framework/Versions/A/Resources/BOSError.strings: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Frameworks/OSInstallerSetup.framework/Versions/A/Frameworks/IAESD.framework/Versions/A/Frameworks/IAMiniSoftwareUpdate.framework/Versions/A/_CodeSignature/CodeResources: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Frameworks/OSInstallerSetup.framework/Versions/A/Frameworks/IAESD.framework/Versions/A/Frameworks/IAMiniSoftwareUpdate.framework/Versions/A/Resources/Info.plist: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Frameworks/OSInstallerSetup.framework/Versions/A/Frameworks/IAESD.framework/Versions/A/Frameworks/IAMiniSoftwareUpdate.framework/Versions/A/Resources/version.plist: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Frameworks/OSInstallerSetup.framework/Versions/A/Frameworks/IAESD.framework/Versions/A/Frameworks/IAPackageKit.framework/Versions/A/_CodeSignature/CodeResources: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Frameworks/OSInstallerSetup.framework/Versions/A/Frameworks/IAESD.framework/Versions/A/Frameworks/IAPackageKit.framework/Versions/A/Resources/Info.plist: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Frameworks/OSInstallerSetup.framework/Versions/A/Frameworks/IAESD.framework/Versions/A/Frameworks/IAPackageKit.framework/Versions/A/Resources/version.plist: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Frameworks/OSInstallerSetup.framework/Versions/A/Frameworks/IAESD.framework/Versions/A/Frameworks/IAInstallerUtilities.framework/Versions/A/_CodeSignature/CodeResources: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Frameworks/OSInstallerSetup.framework/Versions/A/Frameworks/IAESD.framework/Versions/A/Frameworks/IAInstallerUtilities.framework/Versions/A/Resources/Info.plist: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Frameworks/OSInstallerSetup.framework/Versions/A/Frameworks/IAESD.framework/Versions/A/Frameworks/IAInstallerUtilities.framework/Versions/A/Resources/version.plist: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/Info.plist: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/PkgInfo: No space left on device
cp: /Volumes/macOSInstaller/Applications/Install macOS Big Sur.app/Contents/version.plist: No space left on device
nx_kernel_mount:1473: : checkpoint search: largest xid 1652, best xid 1652 @ 27
nx_kernel_mount:1473: : checkpoint search: largest xid 1652, best xid 1652 @ 27
nx_kernel_mount:1473: : checkpoint search: largest xid 1652, best xid 1652 @ 27
nx_kernel_mount:1473: : checkpoint search: largest xid 1652, best xid 1652 @ 27
nx_kernel_mount:1473: : checkpoint search: largest xid 1652, best xid 1652 @ 27
nx_kernel_mount:1473: : checkpoint search: largest xid 1652, best xid 1652 @ 27
nx_kernel_mount:1473: : checkpoint search: largest xid 1652, best xid 1652 @ 27
nx_kernel_mount:1473: : checkpoint search: largest xid 1652, best xid 1652 @ 27
nx_kernel_mount:1473: : checkpoint search: largest xid 1652, best xid 1652 @ 27`

Drive isn't full on the machine, has 257GB available. It's like it's making the DMG too small.

### PR DESCRIPTION
**Pull request by taniguti**
Fixed #160 

- Run this on Mojave, require more space. https://github.com/kc9wwh/macOSUpgrade/pull/161/commits/0380d8a0f34a0188ffe5e90f8edfee044e6b2781
- Converting dmg file on Mojave, put error out. Use JHFS+ instead of APFS. https://github.com/kc9wwh/macOSUpgrade/pull/161/commits/0380d8a0f34a0188ffe5e90f8edfee044e6b2781
- More space on Catalina.  https://github.com/kc9wwh/macOSUpgrade/pull/161/commits/1487bb610cd0aba28b58548b4cff2e6b89321f98

=====
```
ladmin@vmac02:macOSUpgrade/[181]$ sw_vers; df -H
ProductName:    Mac OS X
ProductVersion: 10.14.6
BuildVersion:   18G6042
Filesystem                     Size   Used  Avail Capacity iused               ifree %iused  Mounted on
/dev/disk1s1                   107G    52G    54G    49%  536561 9223372036854239246    0%   /
devfs                          188k   188k     0B   100%     640                   0  100%   /dev
/dev/disk1s4                   107G    20k    54G     1%       0 9223372036854775807    0%   /private/var/vm
map -hosts                       0B     0B     0B   100%       0                   0  100%   /net
map auto_home                    0B     0B     0B   100%       0                   0  100%   /home
.host:/VMware Shared Folders   2.0T   2.0T    40G    99%       0                   0  100%   /Volumes/VMware Shared Folders
ladmin@vmac02:macOSUpgrade/[182]$ ./helper-tools/prepare-jamf-policy.sh /Applications/Inst
Install macOS Big Sur.app/  Install macOS Catalina.app/
ladmin@vmac02:macOSUpgrade/[182]$ ./helper-tools/prepare-jamf-policy.sh /Applications/Install\ macOS\ Big\ Sur.app/
Wait seconds, getting checksum...

=====================================================
Parameters for JamfPro policy:
Parameter 4: /Applications/Install macOS Big Sur.app
Parameter 5: 11.0.1
Parameter 6: (Your download trigger policy)
Parameter 7: b5df443ec0ead5e8daac39fba6bc401e
=====================================================

Would you need a DMG file of Install macOS Big Sur.app? [y/n]: y
Ok, building DMG file of Install macOS Big Sur.app. Wait few minutes.
Done. /Users/ladmin/Downloads/Install macOS Big Sur.app.11.0.1.dmg
ladmin@vmac02:macOSUpgrade/[183]$
```

```
ladmin@vmac06:Documents/macOSUpgrade[195]% sw_vers; df -H
ProductName:    Mac OS X
ProductVersion: 10.15.7
BuildVersion:   19H15
Filesystem                     Size   Used  Avail Capacity iused      ifree %iused  Mounted on
/dev/disk1s5                   107G    11G    76G    13%  488300 1046039300    0%   /
devfs                          191k   191k     0B   100%     650          0  100%   /dev
/dev/disk1s1                   107G    19G    76G    20%  166410 1046361190    0%   /System/Volumes/Data
/dev/disk1s4                   107G   1.1M    76G     1%       1 1046527599    0%   /private/var/vm
.host:/VMware Shared Folders   2.0T   2.0T    46G    98%       0          0  100%   /Volumes/VMware Shared Folders
map auto_home                    0B     0B     0B   100%       0          0  100%   /System/Volumes/Data/home
ladmin@vmac06:Documents/macOSUpgrade[196]% ./helper-tools/prepare-jamf-policy.sh /Applications/Install\ macOS\ Big\ Sur.app
Wait seconds, getting checksum...

=====================================================
Parameters for JamfPro policy:
Parameter 4: /Applications/Install macOS Big Sur.app
Parameter 5: 11.0.1
Parameter 6: (Your download trigger policy)
Parameter 7: b5df443ec0ead5e8daac39fba6bc401e
=====================================================

Would you need a DMG file of Install macOS Big Sur.app? [y/n]: y
Ok, building DMG file of Install macOS Big Sur.app. Wait few minutes.
Done. /Users/ladmin/Downloads/Install macOS Big Sur.app.11.0.1.dmg
ladmin@vmac06:Documents/macOSUpgrade[197]%
```

### CODE REVIEW

**Comment by kenchan0130** [comment]
LGTM

File: helper-tools/prepare-jamf-policy.sh [hunk 1]
```diff
@@ -87,45 +87,57 @@ _RESULT
...
+    dist_dmg="${HOME}/Downloads/$( /usr/bin/basename "$OSInstaller" ).${osversion}.dmg"
+    sizeOfInstaller="$( /usr/bin/du -sm "$OSInstaller" | /usr/bin/awk '{print $1}' )"
     volumename="macOSInstaller"
+    extra_size=512
+    filesystem='APFS'
 
-    /usr/bin/hdiutil create -size $(( sizeOfInstaller + 256 ))m -volname "$volumename" "$temp_dmg" -fs APFS > /dev/null
-    devfile="$( /usr/bin/hdiutil attach -readwrite -nobrowse "$temp_dmg" | awk '$NF == "GUID_partition_scheme" {print $1}' )"
+    if [ "$( /usr/bin/sw_vers -productVersion | /usr/bin/awk -F. '{ print ($1 * 10 ** 2 +  $2 )}' )" -lt 1015 ]; then
+        filesystem='JHFS+'
```
Comment (kenchan0130) [inline helper-tools/prepare-jamf-policy.sh:126]: In mojave, we have to use Mac OS Extended Journaled and that's an error.
Although this project is also guaranteed to work on High Sierra. Does it work?

Comment (kc9wwh) [reply helper-tools/prepare-jamf-policy.sh:126]: @kenchan0130 @taniguti thank you for trying to keep "legacy" support. I was running into this for a PR I am working on submitting this morning. What are your thoughts on using the n-2 methodology? That would have us guaranteeing support for the last 3 major releases. We wouldn't remove support necessarily for older versions, but we also wouldn't regularly test against them. Thoughts?

**Comment by kc9wwh** [comment]
All tests passed on 10.15.5 and 11.0.1. Wasn't able to test on Mojave as my VM is having issues...waiting on re-build. 

```
admin@macOSUpgrade-10 ~ % sw_vers; df -H
ProductName:	Mac OS X
ProductVersion:	10.15.5
BuildVersion:	19F101
Filesystem      Size   Used  Avail Capacity iused      ifree %iused  Mounted on
/dev/disk1s5     86G    11G    43G    21%  487560  836324840    0%   /
devfs           190k   190k     0B   100%     645          0  100%   /dev
/dev/disk1s1     86G    30G    43G    41%   52621  836759779    0%   /System/Volumes/Data
/dev/disk1s4     86G   1.1G    43G     3%       2  836812398    0%   /private/var/vm
map auto_home     0B     0B     0B   100%       0          0  100%   /System/Volumes/Data/home
/dev/disk2s2     13G    12G   366M    98%      15 4294967264    0%   /Volumes/Shared Support
admin@macOSUpgrade-10 ~ % ~/Desktop/prepare-jamf-policy.sh /Applications/Install\ macOS\ Big\ Sur.app 
Wait seconds, getting checksum...

=====================================================
Parameters for JamfPro policy:
Parameter 4: /Applications/Install macOS Big Sur.app
Parameter 5: 11.0.1
Parameter 6: (Your download trigger policy)
Parameter 7: b5df443ec0ead5e8daac39fba6bc401e
=====================================================

Would you need a DMG file of Install macOS Big Sur.app? [y/n]: y
Ok, building DMG file of Install macOS Big Sur.app. Wait few minutes.
Done. /Users/admin/Downloads/Install macOS Big Sur.app.11.0.1.dmg
```

```
josh.roskos@MarvinsessedMBP ~ % sw_vers; df -H
ProductName:	macOS
ProductVersion:	11.0.1
BuildVersion:	20B50
Filesystem       Size   Used  Avail Capacity iused      ifree %iused  Mounted on
/dev/disk1s1s1   500G    15G   258G     6%  563932 4881888948    0%   /
devfs            193k   193k     0B   100%     653          0  100%   /dev
/dev/disk1s4     500G    29G   258G    11%      27 4882452853    0%   /System/Volumes/VM
/dev/disk1s2     500G   281M   258G     1%     766 4882452114    0%   /System/Volumes/Preboot
/dev/disk1s6     500G   918k   258G     1%      13 4882452867    0%   /System/Volumes/Update
/dev/disk1s7     500G   197G   258G    44%  865331 4881587549    0%   /System/Volumes/Data
map auto_home      0B     0B     0B   100%       0          0  100%   /System/Volumes/Data/home
josh.roskos@MarvinsessedMBP ~ % ~/Desktop/prepare-jamf-policy.sh /Applications/Install\ macOS\ Big\ Sur.app 
Wait seconds, getting checksum...

=====================================================
Parameters for JamfPro policy:
Parameter 4: /Applications/Install macOS Big Sur.app
Parameter 5: 11.0.1
Parameter 6: (Your download trigger policy)
Parameter 7: b5df443ec0ead5e8daac39fba6bc401e
=====================================================

Would you need a DMG file of Install macOS Big Sur.app? [y/n]: y
Ok, building DMG file of Install macOS Big Sur.app. Wait few minutes.
Done. /Users/josh.roskos/Downloads/Install macOS Big Sur.app.11.0.1.dmg
```

**Review by kc9wwh — APPROVED** [review]

File: helper-tools/prepare-jamf-policy.sh (hunk 1 above)
Comment (kenchan0130) [reply helper-tools/prepare-jamf-policy.sh:126]: > What are your thoughts on using the n-2 methodology?

I agree with this policy.
However, I just checked the project readme as it was described as supported.

Comment (kc9wwh) [reply helper-tools/prepare-jamf-policy.sh:126]: Yes, it has been and I think we should try, but if it’s gonna cause a lot of additional work then probably not worth it. 

Let’s maintain support for the moment unless we’re hitting a major issue. I’ll reach to to some folks and try and gauge what kind of issues we may run into as far as push back.

Comment (taniguti) [reply helper-tools/prepare-jamf-policy.sh:126]: Yes. I think we must focus recent 3 major releases. Supporting "legacy" itself must make additional tasks and make our code even more complex (and none feels happy).

Thanks @kc9wwh  for checking the script. The output looks good.

Comment (kc9wwh) [reply helper-tools/prepare-jamf-policy.sh:126]: Certainly! Let's just be sure to link to the previous version on the Readme when we drop support for versions so if someone needs to deploy an older version they can go that commit and grab the older version....does that make sense?

**Pull request merged**